src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MagicNumberCheck.java

```diff
@@ -485,26 +485,23 @@ else if (parent.getType() == TokenTypes.UNARY_PLUS) {
     private static boolean isInHashCodeMethod(DetailAST ast) {
         boolean inHashCodeMethod = false;
 
-        // if not in a code block, can't be in hashCode()
-        if (ScopeUtil.isInCodeBlock(ast)) {
```

Comment:
One of things `isInCodeBlock` does is check if the AST is in a method definition. It does not check that any other thing takes priority, like AST could be a field and it will not stop on a class or record. So the next part of the code is always looking for a method definition, which is also what `isInCodeBlock` looks for. Doesn't care if we pass another class definition (anonymous class) to get to said method definition. As long as it finds a method definition, we can check if its a hash code method or not. The code is `null` safe already that if it scans too far it won't NPE.

So there isn't a way to kill this mutation as it is basically duplicating existing code and is null safe.

I think some this might point to an oversight in the check like are we really in a hash code method if its part of a field in the anonymous class inside the hash code method.